src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -112,6 +112,40 @@ protected bool ShouldProcessServiceOperation(ServiceController service)
             WriteError(new ErrorRecord(exception, errorId, category, targetObject));
         }
 
+        internal void SetServiceSecurityDescriptor(
+            ServiceController service,
+            string securityDescriptorSddl,
+            NakedWin32Handle hService)
+        {
+            var rawSecurityDescriptor = new RawSecurityDescriptor(securityDescriptorSddl);
+            RawAcl rawDiscretionaryAcl  = rawSecurityDescriptor.DiscretionaryAcl ;
+            var  discretionaryAcl   = new DiscretionaryAcl (false, false, rawDiscretionaryAcl);
+
+            byte[] rawDacl = new byte[discretionaryAcl.BinaryLength];
+            discretionaryAcl.GetBinaryForm(rawDacl, 0);
+            rawSecurityDescriptor.DiscretionaryAcl = new RawAcl(rawDacl, 0);
+            byte[] securityDescriptorByte = new byte[rawSecurityDescriptor.BinaryLength];
+            rawSecurityDescriptor.GetBinaryForm(securityDescriptorByte, 0);
+
+            bool status = NativeMethods.SetServiceObjectSecurity(
+                        hService,
+                        SecurityInfos.DiscretionaryAcl,
+                        securityDescriptorByte);
```

Comment:
Indentation must be 4 spaces.

Reply:
Thanks, why didn't CodeFactor catch this ?